homeassistant/components/ecodevices/__init__.py

```diff
@@ -0,0 +1,136 @@
+"""GCE Eco-Devices integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pyecodevices import (
+    EcoDevices,
+    EcoDevicesCannotConnectError,
+    EcoDevicesInvalidAuthError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONFIG,
+    CONTROLLER,
+    COORDINATOR,
+    DOMAIN,
+    PLATFORMS,
+    UNDO_UPDATE_LISTENER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Eco-Devices integration."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Eco-Devices from a config entry."""
+    config = entry.data
+
+    session = async_get_clientsession(hass, False)
+
+    controller = EcoDevices(
+        config.get(CONF_HOST),
+        config.get(CONF_PORT),
+        config.get(CONF_USERNAME),
+        config.get(CONF_PASSWORD),
+        session=session,
+    )
+
+    try:
+        await controller.get_info()
+    except EcoDevicesCannotConnectError as exception:
+        raise ConfigEntryNotReady from exception
```

Comment:
In the future, it would be good to implement a reauth flow, where you throw `ConfigEntryAuthFailed` on `EcoDevicesInvalidAuthError`.

Example: https://github.com/home-assistant/core/pull/49413.